### ISSUE TITLE
Maven Plugin: Proxy basic authentication fails with encrypted password, setting jdk.http.auth.tunneling.disabledSchemes="" comes too late

### ISSUE THREAD

**Issue by stefan-prange** [dependency-check/DependencyCheck#2960]
**Describe the bug**

I found (and solved) a problem with proxy authentication when downloading the NVD CVE vulnerability data from the internet over a proxy which requires basic authentication and uses http.

**My environment**
- Windows 10
- OpenJDK 11.0.8
- Maven 3.6.3
- Dependency-Check Maven plugin version 6.0.3
- Internet access is only possible over a proxy which requires basic authentication and which uses the http protocol on port 8080.
- Proxy access data is present as a \<proxies>\<proxy> element in my file \<HOME>/.m2/settings.xml 
- The element \<proxies>\<proxy>\<password> of that file contains an encrypted value, as described on https://maven.apache.org/guides/mini/guide-encryption.html

**Version of dependency-check used**

I'm using the Maven plugin with the following coordinates:
```xml
<plugin>
    <groupId>org.owasp</groupId>
    <artifactId>dependency-check-maven</artifactId>
    <version>6.0.3</version>
</plugin>
```

**Log file**

When running the command "mvn dependency-check:check" on my project, I get the following error information (log snippet):
```txt
Caused by: java.io.IOException: Unable to tunnel through proxy. Proxy returns "HTTP/1.1 407 Proxy Authentication Required"
    at sun.net.www.protocol.http.HttpURLConnection.doTunneling (HttpURLConnection.java:2177)
    at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect (AbstractDelegateHttpsURLConnection.java:183)
    at sun.net.www.protocol.https.HttpsURLConnectionImpl.connect (HttpsURLConnectionImpl.java:168)
    at org.owasp.dependencycheck.utils.HttpResourceConnection.obtainConnection (HttpResourceConnection.java:206)
    at org.owasp.dependencycheck.utils.HttpResourceConnection.fetch (HttpResourceConnection.java:163)
```

**To Reproduce**
1. Set up an environment like mine, as described above, especially with an internet proxy that requires basic authentication and works with http.
2. Run `mvn dependency-check:check` or `mvn dependency-check:update-only` on an arbitrary Maven project in whose POM in its build/plugins or build/pluginManagement/plugins part you added the \<plugin> snippet shown above.

**My solution attempts:**

1.) Define the JVM system property "jdk.http.auth.tunneling.disabledSchemes" with an empty value right on JVM start up (because I knew that to be a possible cause of proxy 407s from earlier cases).
$ set _JAVA_OPTIONS=-Djdk.http.auth.tunneling.disabledSchemes=
Result: still getting the proxy auth 407 error

2.) Provide clear text proxy password in \<HOME>/.m2/settings.xml
Result: still getting the proxy auth 407 error

3.) Apply both measures: clear text proxy password + jdk.http.auth.tunneling.disabledSchemes=""
Result: Downloading the NVD CVE metadata works :-)

**My workaround:**

As mentioned above as solution attempt 3 I had to do the following two things to work around the problem:
1.) Write the proxy password as clear text into the \<proxies>\<proxy>\<password> element in the \<HOME>/.m2/settings.xml file.
2.) Set JVM system property "jdk.http.auth.tunneling.disabledSchemes" with an empty value right on JVM start up:
$ set _JAVA_OPTIONS=-Djdk.http.auth.tunneling.disabledSchemes=

**Expected behavior**
Well, I expected the download of the NVD CVE vulnerability data to just work with my existing proxy definition located in my \<HOME>/.m2/settings.xml file.

**Additional context**

_My expectations on how to fix the issue_

1.) The dependency-check needs to set the JVM property "jdk.http.auth.tunneling.disabledSchemes" to an empty value earlier, maybe as first step of the CLI's main method and as soon as possible in the maven-plugin.
If setting that property by the plugin still comes too late, then the plugin should enforce that the user adds it to his Java resp. 
 Maven environment prior to starting the dependency-check plugin.

2.) The dependency-check Maven plugin should be able to also take over encrypted proxy passwords from the surrounding Maven environment, not only clear-text proxy passwords.

_And an additional note on Maven's password encryption:_

Although https://maven.apache.org/guides/mini/guide-encryption.html describes using encrypted passwords only for server credentials, they can also be used in \<proxy> elements and \<property> values. I'm using that feature in practice on a regular basis.

**Comment by stefan-prange** [comment]
@jeremylong I just wrote a fix for this issue and it works fine with my local build of the dependency-check-maven-plugin, version 6.0.4-SNAPSHOT. I created it based on a copy of the main branch that I cloned just 2 hours ago.
Now I'm not sure how to contribute my fix. Shall I just paste a code diff into a further issue comment? Or would you create a branch and I'll push my changes to it?

**Comment by albuch** [comment]
@stefan-prange: Create a fork of this repository, create a new branch in your forked repository where you apply the fix and then create a pull request to bring your changes back into this repo.

See https://docs.github.com/en/enterprise-server@2.22/github/collaborating-with-issues-and-pull-requests/proposing-changes-to-your-work-with-pull-requests for a detailed walkthorugh.

**Comment by stefan-prange** [comment]
@albuch Thanks for the advice. I'm a first timer in terms of contributing to a Github project and really appreciate your guidance. I just handed in PR #2966 in accordance with it.

**Comment by jeremylong** [comment]
Thanks for the PR - this will be included in 6.0.4 (likely released this weekend).
